src/lexer/tokens/token.rs

```diff
@@ -0,0 +1,40 @@
+use crate::literals::literals::Literal;
+
+use super::tokentype::TokenType;
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct Token {
```

Comment:
this is a pretty complex type that's gonna be a PAIN to move around, try slimming down the amount of fields to the bare minimum for a token, also prob implement a TokenStream type for the final result

Reply:
This is the bare minimum. 
TokenType: To tell what token you are looking at.
Lexeme: Lexemes are used for error messages so would be advisable to keep.
Line: you know...
At: Should've renamed this to column instead would've been better.
Literal: Also obvious.
Its not going to be a pain moving it around trust me.

About the TokenStream type that is a good idea I actually don't know how to implement one will have to look it up.
From what i read seems like an iterator with a couple of more features. (don't quote me)

Reply:
no cause the Token enum can contain the lexeme if any, and if the token is Token::Equals, then well theres no way the lexeme is gonna be anything else so thats redundant. What I'm saying is ye only need a view into the lexeme *sometimes*, amd its better to just wrap it in the enum when needed.

Reply:
But you cant change the whole thing for just a few cases. some places will definitely need the lexeme and there is no avoiding it.

Reply:
Also why are you so worried about optimizing at such an early stage its not like the lexer is going eat all your memory cuz of a few bytes.

Reply:
Look i dont want to pollute the  TokenType Enum its just super messy to do it like this.

Reply:
fine, ya win for now

Reply:
Its not about winning or losing. I just wanted to tell you why i did it the way i did.